File: prime/prime.cs
class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Accept number:");
            int number = Convert.ToInt32(Console.ReadLine());
            if (IsPrime(number))
            {
                Console.WriteLine("It is prime");
            }
            else
            {
                Console.WriteLine("It is not prime");
            }       
        }

        public static bool IsPrime(int number)
        {
            if (number <= 1) return false;
            if (number == 2) return true;
            if (number % 2 == 0) return false;

            var boundary = (int)Math.Floor(Math.Sqrt(number));

            for (int i = 3; i <= boundary; i += 2)
            {
                if (number % i == 0) return false;
            }

            return true;
        }
    }

File: prime/PRime.cpp
#include <iostream>
using namespace std;

int main()
{
  int n, i;
  bool isPrime = true;

  cout << "Enter a positive integer: ";
  cin >> n;
  
  if(n<2){
     cout << "This is not a prime number";
  }
  
  else{
      for(i = 2; i*i<=n; ++i)
      {
          if(n % i == 0)
          {
              isPrime = false;
              break;
          }
      }
      if (isPrime)
          cout << "This is a prime number";
      else
          cout << "This is not a prime number";
  }

  return 0;
}
